bundles/org.openhab.binding.monopriceaudio/src/main/java/org/openhab/binding/monopriceaudio/internal/handler/MonopriceAudioHandler.java

```diff
@@ -0,0 +1,723 @@
+                                logger.warn("Error Turning All Zones Off: {}", e.getMessage());
+                            }
+
+                        });
+                        break;
+                    case CHANNEL_TYPE_ALLSOURCE:
+                        if (command instanceof DecimalType) {
+                            int value = ((DecimalType) command).intValue();
+                            if (value >= ONE && value <= MAX_SRC) {
+                                zoneStream.limit(numZones).forEach((zoneName) -> {
+                                    if (!ignoreZones.contains(zoneName)) {
+                                        try {
+                                            connector.sendCommand(MonopriceAudioZone.valueOf(zoneName),
+                                                    MonopriceAudioCommand.SOURCE, value);
+                                        } catch (MonopriceAudioException e) {
+                                            logger.warn("Error Setting Source for  All Zones: {}", e.getMessage());
+                                        }
+                                    }
+                                });
+                            }
+                        }
+                        break;
+                    case CHANNEL_TYPE_ALLVOLUME:
+                        if (command instanceof PercentType) {
+                            int value = (int) Math
+                                    .round(((PercentType) command).doubleValue() / 100.0 * (MAX_VOLUME - MIN_VOLUME))
+                                    + MIN_VOLUME;
+                            allVolume = value;
+                            zoneStream.limit(numZones).forEach((zoneName) -> {
+                                if (!ignoreZones.contains(zoneName)) {
+                                    try {
+                                        connector.sendCommand(MonopriceAudioZone.valueOf(zoneName),
+                                                MonopriceAudioCommand.VOLUME, value);
+                                    } catch (MonopriceAudioException e) {
+                                        logger.warn("Error Setting Volume for All Zones: {}", e.getMessage());
+                                    }
+                                }
+                            });
+                        }
+                        break;
+                    case CHANNEL_TYPE_ALLMUTE:
+                        MonopriceAudioCommand cmd;
+                        if (command instanceof OnOffType && command == OnOffType.ON) {
+                            cmd = MonopriceAudioCommand.MUTE_ON;
+                        } else if (command instanceof OnOffType && command == OnOffType.OFF) {
+                            cmd = MonopriceAudioCommand.MUTE_OFF;
+                        } else {
+                            cmd = null;
+                        }
+
+                        if (cmd != null) {
```

Comment:
might as well break early instead of having to perform another null check.
```suggestion
                            break;
                        }
```